src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -253,6 +254,157 @@ public static StringValues Concat(StringValues values1, StringValues values2)
             return new StringValues(combined);
         }
 
+        public static bool Equals(StringValues left, StringValues right)
+        {
+            // .Count is an expression not a numeric so cache locally
+            var count = left.Count;
```

Comment:
Might not be but as ```Count => _values?.Length ?? (_value != null ? 1 : 0)``` I didn't want to use it as the test condition on the following for loop. However unlikely to be many values in StringValues so can take it or leave it?